Makefile

```diff
@@ -9,3 +14,26 @@ clean:
 	rm -f .coverage .coverage.*
 	find . -name '*.py[co]' -or -name '*.so*' -or -name '*.dylib' -delete
 	find . -depth -name __pycache__ -exec rm -rf {} \;
+
+# LCOV report
+.PHONY: lcov-clean lcov-coverage lcov-report lcov-open lcov
+lcov-clean:
+	rm -rf $(LCOV_INFO) $(LCOV_REPORT)
+	find build -name '*.gc??' -delete
+
+lcov-coverage:
+	WITH_GCOV=1 tox -e py27,py36
+
+$(LCOV_INFO):
+	lcov --capture --directory build --output-file $(LCOV_INFO)
+
+$(LCOV_REPORT): $(LCOV_INFO)
+	genhtml --output-directory $(LCOV_REPORT) \
+		$(LCOV_REPORT_OPTIONS) $(LCOV_INFO)
+
+lcov-report: $(LCOV_REPORT)
+
+lcov-open: $(LCOV_REPORT)
+	xdg-open $(LCOV_REPORT)/index.html
+
+lcov: lcov-clean lcov-coverage lcov-report
```

Comment:
I don't think make guarantees the order in which it builds dependencies (especially with parallel builds – I have `make` aliased to `make -j4`).

IMO it should be something like:

```
lcov: lcov-clean
    $(MAKE) lcov-coverage
    $(MAKE) lcov-report
```

Reply:
the makefile has ``.NOTPARALLEL:`` to disable all parallel builds. But I can change the code.